Pidgin/Parser.Not.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using Pidgin.ParseStates;
+
+namespace Pidgin
+{
+    public static partial class Parser
+    {
+        /// <summary>
+        /// Creates a parser which attempts to apply the specified parser and only succeeds if that parser fails.
+        /// </summary>
+        /// <param name="parser">The parser to apply if the current parser fails without consuming any input</param>
+        /// <returns>A parser which tries to apply the current parser, and applies <paramref name="parser"/> if the current parser fails without consuming any input.</returns>
+        public static Parser<TToken, TToken> Not<TToken>(Parser<TToken, TToken> parser) {
```

Comment:
I tried going that route, but the problem is that you end up needing a Map function because you don't know how to convert the current token for a successful parse to the return type.